README.md

```diff
@@ -4,3 +4,190 @@ lua-client
 [![Build Status](https://travis-ci.org/neovim/lua-client.svg?branch=master)](https://travis-ci.org/neovim/lua-client)
 
 Lua client for Neovim
+
+
+Basic usage
+-----------
+
+The module `nvim.nvim` provides a set of functions to connect to an instance of neovim.
+
+### Create an instance of the client
+
+First of all we need to get an instance of the client ready to talk with neovim.
+
+#### new_from_socket_file
+
+Connects into a running neovim instance using a socket file stream.
+
+```lua
+local Nvim = require('nvim.nvim')
+local nvim = Nvim.new_from_socket_file('/tmp/.nvim.socket')
+```
+
+#### new_from_socket
+
+Connects into a running neovim instance using a socket tcp stream.
+
+```lua
+local Nvim = require('nvim.nvim')
+local nvim = Nvim.new_from_socket('127.0.0.1', 6666)
+```
+
+#### new_from_process
+
+```lua
+local Nvim = require('nvim.nvim')
+local nvim = Nvim.new_from_process()
+```
+
+This code will create a new embedded neovim instance and will connect into it.
+
+
+### Using the client
+
+#### Listing the types available in the client
+
+Once we have a client we can list the types supported by neovim, the list of types include the basic types and the extended types.
```

Comment:
"Once we have a client, we can list the types supported by neovim. This list includes both basic and extended types."

Or maybe even:

"Once we have a client we can list the basic and extended types supported by neovim."